leakcanary-android-sample/src/main/java/com/example/leakcanary/MainActivity.kt

```diff
@@ -32,6 +41,12 @@ class MainActivity : Activity() {
     val app = application as ExampleApplication
     val leakedView = findViewById<View>(R.id.helper_text)
 
+    // Create a dumb view model that will leak.
+    ViewModelProvider(this, object : Factory {
+      @Suppress("UNCHECKED_CAST")
+      override fun <T : ViewModel?> create(modelClass: Class<T>): T = DumbViewModel() as T
+    }).get(DumbViewModel::class.java)
```

Comment:
Ah, dang it, this single-parameter constructor is not available in `< 2.2.0` version. So the constructor with 2 parameters is the only option that won't be deprecated and that would work in both versions of API.
God, it's an inconsistent API..

Reply:
The ViewModel API has evolved towards a better shape, and the APIs are still backward compatible, so I'd say it's fairly reasonable. `ViewModelProviders` was originally in a different artifact (which now doesn't exist since there is a better and direct way).

Anyway, yes, you just captured the why I'm going with this approach.